keras_nlp/utils/pipeline_model.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A base class for models including preprocessing."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+def convert_inputs_to_dataset(
+    x=None,
+    y=None,
+    sample_weight=None,
+    batch_size=None,
+):
+    if isinstance(x, tf.data.Dataset):
+        if y is not None:
+            raise ValueError(
+                "When `x` is a `tf.data.Dataset`, please do not provide "
+                f"`y`. Received: `y={y}`."
```

Comment:
Might have to truncate or reformat the `y` display because if it's an array that will be a very unreadable message

Reply:
Is there a good way to do this that's safe with a wide range of input types? Just removing for now.

Reply:
Ah seeing comment below, I will print the type!